src/FormBuilder/resources/js/form-builder/src/components/canvas/FormSettings.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {useMemo} from 'react';
+import FormSettingsContainer from '@givewp/form-builder/components/canvas/FormSettingsContainer';
+import FormGeneralSettingsGroup from '@givewp/form-builder/settings/group-general';
+import FormDonationConfirmationSettingsGroup from '@givewp/form-builder/settings/group-donation-confirmation';
+import {__} from '@wordpress/i18n';
+import EmailGeneralSettings from '@givewp/form-builder/settings/group-email-settings/general';
+import getEmailSettings from '@givewp/form-builder/settings/group-email-settings';
+
+/**
+ * @unreleased
+ */
+export default function FormSettings() {
+    const routes: Route[] = useMemo(() => {
+        const additionalSettings = validateRoutes(
+            wp.hooks.applyFilters('givewp_form_builder_settings_additional_routes', [])
+        );
+
+        return [
+            {
+                name: __('General', 'give'),
+                path: 'general',
```

Comment:
Now that the element is being assigned to the FC, let's move this whole routes array outside of the component which would remove the need for useMemo.  Then inside here, we can simply validate the routes array and pass it along to `FormSettingsContainer`

Reply:
Resolved: e44f5683ac19baf68dbe9e8d9f044ae976a49eac